pkg/ring/consul_client.go

```diff
@@ -299,10 +273,10 @@ func (c *consulClient) WatchKey(key string, done <-chan struct{}, f func(interfa
 		})
```

Comment:
I wonder is `Ongoing` should check if `done` is closed, and then replace the call to `isClosed` here with a call to `Ongoing`?

Reply:
From the `Context` docs we would check `Err() != nil`

That would change the semantics; probably in an ok way, but I felt this was conflating the two mechanisms too much and we should check explicitly in the outer loop. 

I have another PR under way where I'm doing that. Maybe I'll change my mind.

Reply:
OK, that comment didn't make very much sense in this context.  See #618 for what I meant.

Reply:
:-) Thanks Brian.  I'll try and look at that tomorrow, but I'm flying then, so may be a little delayed.